pints/tests/test_slice_stepout.py

```diff
@@ -0,0 +1,397 @@
+        self.assertTrue(mcmc._init_narrowing)
+        self.assertFalse(mcmc._init_overrelaxation)
+        self.assertTrue(((mcmc._r - mcmc._l) < 1.1 *
+                         mcmc._w[mcmc._active_param_index]) and
+                        mcmc._init_narrowing)
+
+        # Continue narrowing ``w``
+        x = mcmc.ask()
+        fx = log_pdf.evaluateS1(x)[0]
+        mcmc.tell(fx)
+        self.assertFalse(mcmc._init_narrowing)
+        self.assertFalse(mcmc._set_l_bisection)
+
+        # Now that we have narrowed ``w``, init bisection
+        while mcmc._a_bar > 0:
+            x = mcmc.ask()
+            fx = log_pdf.evaluateS1(x)[0]
+            mcmc.tell(fx)
+
+        # Update parameter and index
+        x = mcmc.ask()
+        fx = log_pdf.evaluateS1(x)[0]
+        self.assertFalse(mcmc._set_l_bisection)
+        self.assertFalse(mcmc._set_r_bisection)
+        mcmc.tell(fx)
+        self.assertEqual(mcmc._active_param_index, 1)
+        self.assertTrue(mcmc._first_expansion)
+        self.assertFalse(mcmc._interval_found)
+
+        """SECOND PARAMETER"""
+        # Expand interval using stepout
+        while not mcmc._interval_found:
+            x = mcmc.ask()
+            fx = log_pdf.evaluateS1(x)[0]
+            mcmc.tell(fx)
+
+        # Check initialisation of overrelaxed step - start narrowing ``w```
+        self.assertTrue(mcmc._init_narrowing)
+        self.assertFalse(mcmc._init_overrelaxation)
+        self.assertFalse(((mcmc._r - mcmc._l) < 1.1 *
+                         mcmc._w[mcmc._active_param_index]) and
+                         mcmc._init_narrowing)
+
+        # Now that we have narrowed ``w``, init bisection
+        while mcmc._a_bar > 0:
+            x = mcmc.ask()
+            fx = log_pdf.evaluateS1(x)[0]
+            mcmc.tell(fx)
+
+        # Update parameter and index
+        x = mcmc.ask()
+        fx = log_pdf.evaluateS1(x)[0]
+        self.assertFalse(mcmc._set_l_bisection)
+        self.assertFalse(mcmc._set_r_bisection)
+        mcmc.tell(fx)
+        self.assertEqual(mcmc._active_param_index, 0)
+        self.assertTrue(mcmc._first_expansion)
+        self.assertFalse(mcmc._interval_found)
+
+    def test_overrelaxed_run(self):
+
+        # Set seed for monitoring
+        np.random.seed(2)
+
+        # Create log pdf
+        log_pdf = pints.toy.GaussianLogPDF([2, 4], [[1, 0], [0, 3]])
+
+        # Create mcmc
+        x0 = np.array([1, 1])
+        mcmc = pints.SliceStepoutMCMC(x0)
+
+        # Set probability of overrelaxed step
+        mcmc.set_prob_overrelaxed(1)
+
+        # Run multiple iterations of the sampler
+        chain = []
+        while len(chain) < 100:
+            x = mcmc.ask()
+            fx = log_pdf.evaluateS1(x)[0]
+            sample = mcmc.tell(fx)
+            if sample is not None:
+                chain.append(np.copy(sample))
+
+        # Fit Multivariate Gaussian to chain samples
+        np.mean(chain, axis=0)
```

Comment:
Again, not sure what these lines are doing?

Reply:
@ben18785 Please refer to the previous comment.